src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1914,7 +1914,17 @@ private AssemblySymbol GetForwardedToAssembly(string fullName, int arity, out bo
 
         internal static void CheckFeatureAvailability(Location location, MessageID feature, DiagnosticBag diagnostics)
         {
-            LanguageVersion availableVersion = ((CSharpParseOptions)location.SourceTree.Options).LanguageVersion;
+            var options = (CSharpParseOptions)location.SourceTree.Options;
```

Comment:
When creating a `CSharpCompilation`, the caller is responsible for ensuring that all trees were constructed using the same language settings (`ArgumentException` is thrown by the constructor if they disagree), and that is the language version assigned to the compilation in the constructor.

I suggest we should remove `features` from `CSharpCompilationOptions` and do the same thing for the features setting.